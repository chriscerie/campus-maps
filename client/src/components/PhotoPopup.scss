#photo-popup-container {
  padding: 0;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
  text-align: center;
  pointer-events: none;
  outline: none;
}

#photo-popup-inner-container {
  display: inline-block;
  height: 100%;
  width: 100%;
  max-width: 960px;
  border-radius: 0.5em;
  box-sizing: border-box;
  background: #fff;
  overflow: hidden;
  transform: translate(0);
  pointer-events: all;
}

#photo-popup-inner {
  display: flex;
  flex-direction: column;
  vertical-align: middle;
  height: 100%;
  width: 100%;
}

#photo-popup-gallery {
  position: relative;
  background-color: #000;
  height: 100%;
  width: 100%;
  cursor: pointer;
  text-align: center;
  vertical-align: middle;
}

#photo-popup-gallery::after {
  content: '';
  display: inline-block;
  vertical-align: middle;
  width: 0;
  height: 100%;
}

#photo-popup-photo {
  display: inline-block;
  vertical-align: middle;
  max-width: 100%;
  max-height: 100%;
  user-select: none;
}

#photo-popup-photo-background {
  position: absolute;
  background: linear-gradient(rgba(0, 0, 0, 0) 95%, rgba(0, 0, 0, 0.8) 100%);
  background-color: transparent;
  padding: 0.2em;
  vertical-align: bottom;
  inset: 0;
  overflow: hidden;
  border-radius: 0.3em;
}

#photo-popup-photo-background #text-area {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 0.6em;
  color: #ddd;
  text-align: center;
  font-size: 0.8em;
}

#photo-popup-sidebar-container {
  flex: 0 0 auto;
  width: auto;
  padding: 1em;
}

@media (min-width: 600px) {
  #photo-popup-inner {
    flex-direction: row;
  }

  #photo-popup-sidebar-container {
    width: 15em;
  }
}
