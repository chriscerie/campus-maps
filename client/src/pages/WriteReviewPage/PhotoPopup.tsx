import Modal from '@mui/material/Modal';
import { useState } from 'react';
import CloseButton from '../../components/CloseButton';
import './PhotoPopup.scss';

function PhotoPopup({
  photos,
  index,
  setIndex,
  onClose,
}: {
  photos: Array<{ imageSrc: string; file: File }>;
  index: number;
  setIndex: React.Dispatch<React.SetStateAction<number>>;
  onClose: (event: {}) => void;
}) {
  const [imageMousePos, setImageMousePos] = useState<'Left' | 'Right'>('Right');

  return (
    <Modal
      open={index >= 0}
      onClose={onClose}
      sx={{
        padding: '3em 1em',
        '@media (min-width: 769px)': {
          padding: '3em 2em',
        },
      }}
    >
      <div id="photo-popup-container">
        <div id="photo-popup-inner-container">
          <CloseButton
            sx={{
              position: 'absolute',
              right: '1em',
              top: '1em',
            }}
            onClick={() => {
              onClose({});
            }}
          />
          <div id="photo-popup-inner">
            <div
              id="photo-popup-gallery"
              onMouseMove={(e) => {
                const posX =
                  e.clientX - e.currentTarget.getBoundingClientRect().left;
                const isLeftSide = posX < e.currentTarget.clientWidth / 2;
                setImageMousePos(isLeftSide ? 'Left' : 'Right');
              }}
              onClick={() => {
                imageMousePos === 'Left'
                  ? setIndex((prev) => Math.max(0, prev - 1))
                  : setIndex((prev) => Math.min(photos.length - 1, prev + 1));
              }}
            >
              <div id="photo-popup-photo-background">
                <div id="text-area">
                  <pre style={{ margin: 0, userSelect: 'none' }}>
                    {imageMousePos === 'Left' &&
                      `<  ${index + 1} of ${photos.length}   `}
                    {imageMousePos === 'Right' &&
                      `   ${index + 1} of ${photos.length}  >`}
                  </pre>
                </div>
              </div>
              <img
                id="photo-popup-photo"
                alt="Location"
                src={photos[index] && photos[index].imageSrc}
              />
            </div>
            <div id="photo-popup-sidebar-container"></div>
          </div>
        </div>
      </div>
    </Modal>
  );
}

export default PhotoPopup;
