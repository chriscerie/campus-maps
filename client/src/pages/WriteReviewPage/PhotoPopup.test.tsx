import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../reducers';
import PhotoPopup from './PhotoPopup';

describe('WriteReviewPage', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <PhotoPopup
              photos={[]}
              index={0}
              setIndex={() => {}}
              onClose={() => {}}
            />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
