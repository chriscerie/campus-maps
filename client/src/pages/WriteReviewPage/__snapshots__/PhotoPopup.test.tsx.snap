// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WriteReviewPage should render successfully 1`] = `null`;
